agent/agentcontainers/api.go

```diff
@@ -1280,12 +1303,51 @@ func (api *API) maybeInjectSubAgentIntoContainerLocked(ctx context.Context, dc c
 		)
 
 		// Create new subagent record in the database to receive the auth token.
+		// If we get a unique constraint violation, try with expanded names that
+		// include parent directories to avoid collisions.
 		client := *api.subAgentClient.Load()
-		newSubAgent, err := client.Create(ctx, subAgentConfig)
-		if err != nil {
-			return xerrors.Errorf("create subagent failed: %w", err)
+
+		originalName := subAgentConfig.Name
+		maxAttempts := 5
+
+		for attempt := 1; attempt <= maxAttempts; attempt++ {
```

Comment:
Do we know all sub-agent names for the workspace at this point? If so, could we not do this check in-memory before trying the creation?

Reply:
Most workspaces are going to only have a single agent. It makes sense to avoid this round trip if we can.